packages/components/src/components/dialog/dialog.jsx

```diff
@@ -6,126 +6,128 @@ import { CSSTransition } from 'react-transition-group';
 import Button from '../button/button.jsx';
 import Icon from '../icon/icon.jsx';
 
-class Dialog extends React.Component {
-    componentDidMount() {
-        if (this.props.is_visible && this.props.disableApp) {
-            this.props.disableApp();
+const Dialog = ({
+    disableApp,
+    enableApp,
+    is_closed_on_cancel,
+    is_closed_on_confirm,
+    is_visible,
+    onCancel,
+    onConfirm,
+    ...other_props
+}) => {
+    React.useEffect(() => {
+        if (is_visible && disableApp) {
+            disableApp();
         }
-    }
+    }, []);
```

Comment:
this is not required. `useEffect` below this should be enough

Reply:
Thanks, you're right, the second call was already enough